sdk/src/main/java/io/opentelemetry/sdk/trace/TracerSdk.java

```diff
@@ -67,7 +71,11 @@ public Scope withSpan(Span span) {
   }
 
   @Override
-  public void recordSpanData(SpanData span) {}
+  public void recordSpanData(SpanData spanData) {
+    synchronized (this) {
+      readableSpans.add(ReadableSpanData.create(spanData));
```

Comment:
I would simply do 
`activeSpanProcessor.onEndSync(ReadableSpanData.create(spanData));`

Maybe also add a TODO for the moment in the SpanProcessor definition to understand if we need to know onEndSync if the start was called or not.

Reply:
Make sense, updated and added a TODO.